### DISCUSSION TITLE
Export dialog UX enhancements

### DISCUSSION THREAD

**Discussion by novhack** [Orama-Interactive/Pixelorama#605] [Ideas]
From recent discussions on Discord it seems that the current export dialog is not very user friendly.

I propose that the dialog could have three export tabs:
- Image
- Spreadsheet
- Animation

I suggest that every tab should have a "From: " selector with these options:
- Image
    - Single frame, All frames, All tags, Tag 1, Tag 2, ..., Tag N
    - When "Single frame" is selected a "Frame:" option should pop up and allow user to select one frame
    - When tag options are selected a "Create new directory for each frame tag" check box should appear (this basically moves this functionality from the current Animation tab)
- Spritesheet
    - All frames, All tags, Tag 1, Tag 2, ..., Tag N
    - All tags have a new check box "As separate files:". If selected all tags will be separate spritesheets otherwise they are in one.
    - All options enable "Orientation:" and "Columns:" properties
    - Possible new option "Gap:" that will add a border between cells of the spreadsheet. By default 0px. (I've seen some spreadsheets distributed with borders around frames)
- Animation:
    - Identical options to Spritesheet
    - "Direction:" selector displayed for all options

My idea behind this proposal is that if user decides to export an animated GIF the dialog workflow could work like this:
The title of the dialog is "Export...". User selects "Animation" tab. Then he select an option from "From:" selector - for instance "All Tags". The final flow would then be "Export... Animation From: All Tags."

I also propose that we should remove the "Interpolation:" option. In the 99.9% of cases users will export pixel art with nearest interpolation and giving them more options that might not even be good for pixel art seems to me as not UX friendly.

It would be also nice if last export options would be stored directly in .pxo files.

**Comment by novhack** [comment]
These proposals are also related to Export dialog:
- #585
- #472

**Comment by Erevoid** [comment]
I mostly agree to this proposal and I can suggest how to update it even more. First of all the term "directory" could be switched with "folder" to be even more familiar to most (I know it's already called directory, I'm suggesting a change to the original name). Secondly instead of the options "Single frame, All frames, All tags, Tag 1, Tag 2, ..., Tag N" we could have "Single frame, All frames, All tags, Selected tags" and if the user chooses the "Selected Tags" more options pop up, showing the user checkboxes with Tag 1, Tag 2, ..., Tag N. This way, if let's say we have 10 tags the user will be able to export from 1 to 10 tags instead of either 1 or 10. It could even have a checkbox to contain all the frames outside of the tags. Lastly, we could keep the different interpolation methods under a category called "advanced options" to prevent people who don't know what they're doing from changing it. Oh and the Animation tab could be called Animation (GIF) if from now on it only exports to GIFs (until we add more exporting options at least)

**Comment by OverloadedOrama** [comment]
I would recommend having two tabs instead of three. "Frame" and "Animation" can be unified into a single tab while keeping "Spritesheet" separate. The new unified tab would export all frames as separate png files by default, with the option to export only the selected frame(s), thus keeping the previous functionality of the "Frame" tab of only exporting a single frame. To export gifs and apngs, we could simply change the file extension.

![image](https://user-images.githubusercontent.com/35376950/202930207-db8e4807-9ba9-41e0-87b8-87f17d31a294.png)

The spritesheet tab can have the exact same options, with extra options added such as the orientation, columns, and gap in the future.

![image](https://user-images.githubusercontent.com/35376950/202930262-5a5285eb-84c8-475a-abc9-28e885bdf86f.png)

I've also added the ability to export selected layers instead of all of the visible ones, which _kind of_ addresses #773. If this is desirable, I will open a PR soon to merge it (EDIT: #781). In the future, we could add extra options under an advanced category, whose elements will be collapsed by default, and interpolation and "Create new folder for each frame tag" could be moved there.
